From 92d3437e4e7179364519da8e9d30ed5cc29feb12 Mon Sep 17 00:00:00 2001
From: Osamu Aoki <osamu@debian.org>
Date: Sat, 8 Feb 2014 16:54:20 +0900
Subject: [PATCH] pass multiarch path to -DLIBDIR

This is approach actually used to package ibus-qt.

== pros ==

It works for me :-)  (ibus-qt)

This approach actually comes from :
 * how upstream CMakefile.txt sets default path
 * how rpm package sets installation path

In CMakefile.txt:
 Define variables
if(NOT LIBDIR)
    if( $ENV{MACHTYPE} MATCHES "64")
        set(LIBDIR lib64)
    else()
        set(LIBDIR lib)
    endif()
endif(NOT LIBDIR)

In ibus-qt.spec.in:

%build
%cmake \
    -DCMAKE_INSTALL_PREFIX=%{_usr} \
    -DLIBDIR=%{_libdir}

== cons ==

This LIBDIR aproach does not seem to be official way to set it.  It seems
something internal to CMake.
---
 Debian/Debhelper/Buildsystem/cmake.pm | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Debian/Debhelper/Buildsystem/cmake.pm b/Debian/Debhelper/Buildsystem/cmake.pm
index db971db..3a0888a 100644
--- a/Debian/Debhelper/Buildsystem/cmake.pm
+++ b/Debian/Debhelper/Buildsystem/cmake.pm
@@ -42,6 +42,7 @@ sub configure {
 
 	# Standard set of cmake flags
 	push @flags, "-DCMAKE_INSTALL_PREFIX=/usr";
+	push @flags, "-DLIBDIR=/usr/lib/$multiarch";
 	push @flags, "-DCMAKE_VERBOSE_MAKEFILE=ON";
 	push @flags, "-DCMAKE_BUILD_TYPE=None";
 
-- 
1.9.rc1

